### PR TITLE
lib: .pseudoRandom(min, max)

### PR DESCRIPTION
**Pull request by Ianis**
Add the possibility to generate big random numbers.

### CODE REVIEW

File: lib/bn.js [hunk 1]
```diff
@@ -2402,4 +2402,28 @@ Mont.prototype.invm = function invm(a) {
...
+//
+
+BN.random = function random(min, max) {
+  var range = max.sub(min);
+  var out = new BN(null);
+  out.length = range.length;
+  out.words = [];
+
+  for (var i = range.length - 1, restricted = true; i >= 0; i--) {
+    var top = restricted ? range.words[i] : 1e7;
```
Comment (indutny) [inline lib/bn.js]: What is `1e7` here?

File: lib/bn.js [hunk 2]
```diff
@@ -2402,4 +2402,28 @@ Mont.prototype.invm = function invm(a) {
...
 
+//
+// Random number generator
+//
+
+BN.random = function random(min, max) {
+  var range = max.sub(min);
+  var out = new BN(null);
+  out.length = range.length;
+  out.words = [];
```
Comment (indutny) [inline lib/bn.js]: Let's create an array and fill it. Unshifting is very slow.

**Comment by Ianis** [comment]
It's actually an old code and I just noticed big mistakes. 
The `1e7 ` was supposed to fix it but it make it worse in some specific conditions.

**Comment by dcousens** [comment]
I think you should be very careful about this,  given this libraries predominant usage in cryptography applications.

Perhaps `insecureRandom`?

**Comment by indutny** [comment]
@dcousens pseudoRandom?

**Comment by dcousens** [comment]
@indutny sure, but it is a very poor `pseudoRandom`.

**Comment by Ianis** [comment]
@dcousens Yes indeed Furthermore JavaScript's `Math.random()` function is not considered a cryptographically-secure random number generator.

**Comment by indutny** [comment]
@dcousens makes sense... do you think it might be a better idea to release it as a separate module instead?

**Comment by dcousens** [comment]
@indutny if it is not to be inlined, then you can easily just use the `randombytes` module that already exists and just import it as hex.

**Comment by Ianis** [comment]
@dcousens  You mean it should support `crypto.randomBytes()` and `window.crypto.getRandomValues()` depending on whether it is running in node or in a browser ?

**Comment by indutny** [comment]
I think it should go into separate module...

**Comment by indutny** [comment]
Sorry!

**Pull request closed without merging**